### PR TITLE
Use current mouse button state instead of saved values.

### PR DESCRIPTION
**Pull request by bruvzg**
Same as https://github.com/godotengine/godot/pull/92415, but for mouse buttons.

### CODE REVIEW

File: platform/linuxbsd/x11/display_server_x11.cpp [hunk 1]
```diff
@@ -4747,12 +4761,20 @@ void DisplayServerX11::process_events() {
 				} else if (mb->get_button_index() == MouseButton::MIDDLE) {
 					mb->set_button_index(MouseButton::RIGHT);
 				}
-				mb->set_button_mask(_get_mouse_button_state(mb->get_button_index(), event.xbutton.type));
 				mb->set_position(Vector2(event.xbutton.x, event.xbutton.y));
 				mb->set_global_position(mb->get_position());
 
 				mb->set_pressed((event.type == ButtonPress));
 
+				if (mb->is_pressed() && mb->get_button_index() >= MouseButton::WHEEL_UP || mb->get_button_index() <= MouseButton::WHEEL_RIGHT) {
```
Comment (AThousandShips) [inline platform/linuxbsd/x11/display_server_x11.cpp:4769]:
```suggestion
				if (mb->is_pressed() && mb->get_button_index() >= MouseButton::WHEEL_UP && mb->get_button_index() <= MouseButton::WHEEL_RIGHT) {
```
I'm assuming

**Review by deralmas — APPROVED** [review]
Gave a look, things look fine!

I think that the freelook fix bundled is more of a workaround (see https://github.com/godotengine/godot/pull/92455#issuecomment-2138068102 ) but it will do for now.

I also researched a bit on the Wayland side and things aren't bad at all. There's a bit of a focus tracking bug in the Wayland thread (which I'll fix soon), but otherwise everything gets reported properly by the compositor even in between focus changes (rightly so, as there's no pointer query API on WL), so we're mostly set :D

**Comment by bruvzg** [comment]
> I think that the freelook fix bundled is more of a workaround

It is, and it's not related to mouse button state. Freelook can be activated by keyboard as well, and it's only checking its internal state after it is activated (or release event is received), releasing it on focus out as well seems to be reasonable thing to do.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**